tensorflow/lite/kernels/conv.cc

```diff
@@ -142,13 +143,66 @@ void TransposeFloatTensor(TfLiteTensor* input, TfLiteTensor* output) {
   }
 }
 
+// Check if im2col needs to be allocated, as some version of optimized Conv dont
+// use it. If any change is supporting im2col in any of the Conv versions, then
+// it should be updated here as well
+bool IsIm2ColRequired(TfLiteTensor* input, TfLiteConvParams* params,
+                      TfLiteTensor* filter, OpData* data, bool is_hybrid,
+                      KernelType kernel_type) {
+  // If HWCN weights are required, Im2Col not required
+  if (data->need_hwcn_weights) return false;
+
+  // segregate based on dilated conv & non-dialated conv
+  const bool need_dilated_im2col =
+      params->dilation_width_factor != 1 || params->dilation_height_factor != 1;
+  const bool need_non_dilated_im2col =
+      params->stride_width != 1 || params->stride_height != 1 ||
+      filter->dims->data[2] != 1 || filter->dims->data[1] != 1;
+
+  const bool need_im2col = need_dilated_im2col || need_non_dilated_im2col;
+
+  // Return early as basic requirement is not met
+  if (!need_im2col) return false;
+
+  // Special case for Hybrid, as it supports only non-dilated im2col currently
+  const bool is_hybrid_non_dilated = is_hybrid && need_non_dilated_im2col;
+  const bool is_quantized =
+      input->type == kTfLiteUInt8 || input->type == kTfLiteInt8;
+
+  switch (kernel_type) {
+    case kReference:
+      if (is_hybrid) {
+        return true;
+      } else {
+        return false;
+      }
+    case kGenericOptimized:
+    case kCblasOptimized:
+      if (is_hybrid && !need_non_dilated_im2col) {
+        return false;
+      } else {
+        return true;
+      }
+    case kMultithreadOptimized:
+      if (is_hybrid_non_dilated || is_quantized ||
+          !data->supports_multithreaded_kernel) {
+        return true;
+      } else {
+        return false;
+      }
+    default:
+      return false;
+  }
+}
+
 // Allocate temporary tensors (`im2col`, `hwcn_weights` if necessary).
 // Note: `context->AddTensors` might invalidate pointers to existing tensors.
 // Therefore the logic to add tensors are isolated into this function.
 static TfLiteStatus AllocateTemporaryTensorsIfRequired(TfLiteContext* context,
                                                        TfLiteNode* node,
                                                        bool is_hybrid,
-                                                       bool is_per_channel) {
+                                                       bool is_per_channel,
+                                                       KernelType kernel_type) {
   auto* params = reinterpret_cast<TfLiteConvParams*>(node->builtin_data);
   OpData* data = reinterpret_cast<OpData*>(node->user_data);
 
```

Comment:
You can remove the filter_width/filter_height vars on lines 213, looks like those are currently breaking the build.